dataflux_pytorch/multipart_upload/multipart.py

```diff
@@ -0,0 +1,309 @@
+                    url,
+                    upload_id,
+                    view,
+                    start=start,
+                    end=end,
+                    part_number=part_number,
+                    checksum=checksum,
+                    headers=headers,
+                    retry=retry,
+                ))
+
+        concurrent.futures.wait(futures,
+                                timeout=deadline,
+                                return_when=concurrent.futures.ALL_COMPLETED)
+
+    try:
+        # Harvest results and raise exceptions.
+        for future in futures:
+            part_number, etag = future.result()
+            container.register_part(part_number, etag)
+
+        container.finalize(blob._get_transport(client))
+    except Exception:
+        container.cancel(blob._get_transport(client))
+        raise
+
+
+class _BufferViewXMLMPUPart(XMLMPUPart):
+
+    def __init__(
+        self,
+        upload_url,
+        upload_id,
+        view,
+        start,
+        end,
+        part_number,
+        headers=None,
+        checksum=None,
+    ):
+        super().__init__(upload_url, upload_id, None, start, end, part_number,
+                         headers, checksum)
+        self._view = view
+
+    def _prepare_upload_request(self):
+        """Prepare the contents of HTTP request to upload a part.
+
+        This is everything that must be done before a request that doesn't
+        require network I/O. This is based on the `sans-I/O`_ philosophy.
+
+        For the time being, this **does require** some form of I/O to read
+        a part from ``stream`` (via :func:`get_part_payload`). However, this
+        will (almost) certainly not be network I/O.
+
+        Returns:
+            Tuple[str, str, bytes, Mapping[str, str]]: The quadruple
+
+              * HTTP verb for the request (always PUT)
+              * the URL for the request
+              * the body of the request
+              * headers for the request
+
+            The headers incorporate the ``_headers`` on the current instance.
+
+        Raises:
+            ValueError: If the current upload has finished.
+
+        .. _sans-I/O: https://sans-io.readthedocs.io/
+        """
+        if self.finished:
+            raise ValueError("This part has already been uploaded.")
+
+        MPU_PART_QUERY_TEMPLATE = "?partNumber={part}&uploadId={upload_id}"
+
+        payload = bytes(self._view[self._start:self._end])
+
+        self._checksum_object = _helpers._get_checksum_object(
+            self._checksum_type)
+        if self._checksum_object is not None:
+            self._checksum_object.update(payload)
+
+        part_query = MPU_PART_QUERY_TEMPLATE.format(part=self._part_number,
+                                                    upload_id=self._upload_id)
+        upload_url = self.upload_url + part_query
+        return "PUT", upload_url, payload, self._headers
+
+
+def _buffer_view_upload_part(
+    maybe_pickled_client,
```

Comment:
I forgot to do this earlier, but we can change this to "client", delete the "maybe_picked_client" documentation in the docstring, and cut the entire line 293-296 code block that sets client. There's no need for it because maybe_pickled_client is for multiprocessing support only.